File: 513. Find Bottom Left Tree Value.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func findBottomLeftValue(root *TreeNode) int {
    answer := 0
    queue := []*TreeNode{}
    queue = append(queue, root)
    for len(queue) > 0 {
        current := queue[0]
        queue = queue[1:]
        answer = current.Val
        if current.Right != nil {
            queue = append(queue, current.Right)
        }
        if current.Left != nil {
            queue = append(queue, current.Left)
        }
    }
    return answer
}

File: 513. Find Bottom Left Tree Value.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int findBottomLeftValue(TreeNode root) {
        int answer = 0;
        Deque<TreeNode> bfsQ = new ArrayDeque<>();
        bfsQ.addLast(root);
        while (!bfsQ.isEmpty()) {
            TreeNode current = bfsQ.pollFirst();
            answer = current.val;
            if (current.right != null) bfsQ.addLast(current.right);
            if (current.left != null) bfsQ.addLast(current.left);
        }
        return answer;
    }
}
